lang/java/reef-wake/wake/src/test/java/org/apache/reef/wake/test/remote/TransportHttpTest.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.wake.test.remote;
+
+import org.apache.reef.tang.Injector;
+import org.apache.reef.tang.Tang;
+import org.apache.reef.tang.exceptions.InjectionException;
+import org.apache.reef.wake.EStage;
+import org.apache.reef.wake.impl.LoggingUtils;
+import org.apache.reef.wake.impl.TimerStage;
+import org.apache.reef.wake.remote.Codec;
+import org.apache.reef.wake.remote.address.LocalAddressProvider;
+import org.apache.reef.wake.remote.impl.ObjectSerializableCodec;
+import org.apache.reef.wake.remote.impl.TransportEvent;
+import org.apache.reef.wake.remote.transport.Link;
+import org.apache.reef.wake.remote.transport.Transport;
+import org.apache.reef.wake.remote.transport.TransportFactory;
+import org.apache.reef.wake.remote.transport.netty.MessagingTransportFactory;
+import org.apache.reef.wake.remote.transport.netty.LoggingLinkListener;
+import org.apache.reef.wake.test.util.Monitor;
+import org.apache.reef.wake.test.util.TimeoutHandler;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestName;
+
+import java.net.InetSocketAddress;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.logging.Level;
+
+import static org.apache.reef.wake.remote.transport.TransportFactory.ProtocolTypes;
+
+
+/**
+ * Tests for Transport via http.
+ */
+public class TransportHttpTest {
```

Comment:
Could you please refactor the test codes? you can use the same test cases by injecting different netty transports

Reply:
I'm very sorry but I cannot fully understand the saying 'injecting different netty transports'. Could you kindly please explain the direction I should refactor in more detail?

Reply:
There are lots of duplicate codes in `TransportHttpTest` and `TransportTest`, so I thought we need to refactor it. 
But, it is an separate issue, so let's refactor it in another issue :) Could you please create an issue for it ?